MongoDB/node.aggregate.js

```diff
@@ -0,0 +1,27 @@
+const _ = require('lodash');
+let logger = global.logger;
+
+module.exports = async (connectorData, inputData) => {
+    logger.trace("MongoDB Aggregate Node: Invoked!");
+    try {
+        logger.trace(`MongoDB Aggregate Node: Input - ${JSON.stringify(inputData)}`);
+
+        if (_.isEmpty(inputData.pipeline)) {
+            throw new Error('Pipeline is empty');
+            logger.error('MongoDB Aggregate Node: Pipeline is empty');
```

Comment:
This line will never be called as you are already throwing an exception before it. Logger should be called before you throw the error.